### DISCUSSION TITLE
Completely unable to configure Talos in a Proxmox VM

### DISCUSSION THREAD

**Discussion by OxygenCobalt** [siderolabs/talos#9291] [Q&A]
Followed the instructions [here](https://www.talos.dev/v0.13/virtualized-platforms/proxmox) to create a VM for Talos inside a fresh Proxmox installation, and for some reason simply nothing about the installation is working.

1. The installation is constantly repeating these lines about failing to resolve Cloudflare's time server, despite the overall Proxmox installation having internet access, and the VM being connected to the correct wireless bridge. The VM even indicates that it's "connected" and is still accessible over the network from my main machine.
 
![image](https://github.com/user-attachments/assets/84d76895-538d-4a70-94d5-6b33647ea62c)
![image](https://github.com/user-attachments/assets/9b5870a4-8d64-4413-a749-0b57ac160f1f)

My network configuration:
![image](https://github.com/user-attachments/assets/4e63e005-2266-402e-87cf-a540f3ea8b83)

![image](https://github.com/user-attachments/assets/f0a3af7f-5a11-459f-8e0c-9cf6cca4463b)

(I configure my server on a private network)

2. When I try to generate and apply a config using [this](https://www.talos.dev/v1.7/talos-guides/install/virtualized-platforms/proxmox/#generate-machine-configurations), the command "succeeds" only for the Talos VM to enter an "Installing" state indefinitely. It never reboots or returns to Maintenance. All further attempts to use `talosctl` are refused. A second network resolution error also appears:

![image](https://github.com/user-attachments/assets/45d8ccdc-8e31-44b1-abef-d93c8ce4d3c2)

Strangely, the cloudflare error also changes
![image](https://github.com/user-attachments/assets/16910e56-7548-496a-b6a4-01ec21075c86)

3. If I try to bootstrap the talos VM (something I've seen in similar issues), it also doesn't work. I can't do it after the apply-config due to it rejecting the connections as said prior, and I can't do it before either due to it seemingly rejecting the authentication with:

`error executing bootstrap: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: tls: failed to verify certificate: x509: certificate signed by unknown authority"`

4. Trying to manually configure the IP assuming no DHCP server (from [here](https://www.talos.dev/v1.7/talos-guides/install/virtualized-platforms/proxmox/#without-dhcp-server)) doesn't work *either*. It just causes the VM to explicitly state it is disconnected while still repeating the same connection errors. It's also no longer accessible externally.

5. I have disabled all firewalls in my Proxmox instance to see if it was destroying any packets, still nothing changed.

I'm frankly just stumped at this point. Can anyone help?

**Comment by smira** [comment]
`network is unreachable` is one way or another problem with your default gateway or routing. 

instead of screenshots, please grab console (serial) logs from Proxmox, and look around the networking setup (or post the logs here)

**Reply by OxygenCobalt** [reply]
Where do I find that? Sorry, I'm new to Proxmox

**Reply by smira** [reply]
I have never used Proxmox at all, so I can't help with it.

**Comment by sakers** [comment]
I'm struggling with this too following the same instructions. Networking is fine and other VMs on the same network bridge can access the internet perfectly well.

**Reply by smira** [reply]
It's very hard to guess what is wrong without any data. 

If you can talk to Talos API, use `talosctl support` to create a bundle and upload it here.

**Reply by sakers** [reply]
![image](https://github.com/user-attachments/assets/44476102-ef40-4718-9550-abeee228461b)
![image](https://github.com/user-attachments/assets/504c439b-411c-437c-9d60-af506caa2171)

No luck with the talosctl support command I'm afraid.

**Reply by smira** [reply]
Please see https://www.talos.dev/v1.7/introduction/getting-started/#default-talosconfig-configuration-file

But you also you haven't bootstrapped your cluster: https://www.talos.dev/v1.7/introduction/getting-started/#kubernetes-bootstrap

Probably reading that getting started document would help.

**Reply by sakers** [reply]
Cheers. 

That's helped. Finally got the first node up and running.

**Comment by kastl-ars** [comment]
I found that triggering a "real" reboot (instead of using kexec) makes the actual reboot more easily recognizable.

I also found that any manual IP configuration (set via `ip=...` when booting off the ISO or image) was not respected with a kexec reboot. But when set in the machine configuration it was respected after a "full" reboot.

**Reply by smira** [reply]
That is totally expected, as after install the kernel command line is rebuilt completely. The `ip=` arg is only used for the initial network configuration. Talos supports [`nocloud`](https://www.talos.dev/v1.7/talos-guides/install/cloud-platforms/nocloud/) when on Proxmox

**Reply by kastl-ars** [reply]
I know, I just wanted to point this out to the OP (as it is not super self-explanatory that there is a kexec-based reboot that is not being shown on any "physical" terminal (at least I could never see it on my machines).

**Comment by rkerno** [comment]
Guys, I had very similar issues getting my first Talos cluster up and running.  I was attempting to configure a 3 node control plane.  I found that setting the IP address through GRUB or editing using the F3 console key both were flakey, with similar symptoms to what you have described.  Using GRUB or F3 I would end up with weird situations where some of the Talos installation used the static I was trying to assign, but other parts seemed to have used a random DHCP address.  It consistently ended with nothing listening on port 6443, and unfortunately, all of the linked troubleshooting relies on port 6443 being listened to.  In the end I configured DHCP with my desired static IP addresses to support the installation, with the cluster address configured in DNS with each of the three IPs configured within DHCP.  The install guides would benefit by including reference to the troubleshooting capabilities of talosctl such as dmesg, cat and ls.

**Reply by bradennapier** [reply]
Duck

**Reply by jfmatth** [reply]
The only static I have in my configuration is for the VIP, see below.  The rest of my Proxmox VM's are DHCP, which makes building the c,luster super easy.

Turns out that Cilium integrates nicely with this configuration too.

If you end up doing static, then you might think of doing all CP nodes with static.

**Comment by jfmatth** [comment]
I use Proxmox and Talos and it's so easy.  

I use a controlplane-patch yaml with a VIP so that all my hosts can just use DHCP, which makes building so much easier
```
machine:
  network:
    interfaces:
    - deviceSelector:
        physical: true
      dhcp: true
      vip:
        ip: 192.168.100.140    # # `interfaces` is used to define the network interface configuration.
```

When I gen my config, I use the VIP IP so that the talos controlplane.yaml is correct.

All of this is in a single batch file from powershell :)

I have my CP VM's set to 2x2, DHCP and boot the ISO for Talos.  Once they boot, I use my batch file to boot to the IP of the VM.  Once Talos is installed and bootstrapped, the VIp is asssigned and just adding 'roles' is easy.  The other CP's are then just added with the addrole.bat file to basically apply the controlplane.yaml to the DHCP address by proxmox.  The same process is for workers.

I can bring up a 5 node Talos cluster in < 10m with all my files.  (3 x CP + 2 x WK).  Happy to help where i can.

Feel free to checkout my talos stuff here... https://github.com/jfmatth/home-charts/tree/cilium-gateway/talos

I'm currently trying to get Cilium working with gateway api and inline yaml, uggh.

**Reply by chamathjayasekara99** [reply]
Is it okay to setup talos on Proxmox vm for production ready cluster?

**Reply by smira** [reply]
Yes, many users do that.

**Comment by chamathjayasekara99** [comment]
Thank you for your response, BTW did you use BMC like phoenixNap to set this  up?

**Comment by angegar** [comment]
I have the same kind of issue here ;)

It is my first time with Talos. I try to deploy a simple VM on Proxmox. The VMs never get the DHCP IP while with the same network configuration other OS can retrieve the IP.

Do you have a sample configuration (the one from the doc start to be old). I just want to ensure that the issue does not come from Talos itself.

FYI: I have a dhcp error in the logs: unable to receive an offer: got an error while the discovery request, no matching response packet received link eth0

**Comment by chamathjayasekara99** [comment]
Can you share screenshot of NIC firewall , VM firewall configuration?
